tests/TestCase/ORM/RulesCheckerIntegrationTest.php

```diff
@@ -490,6 +490,28 @@ public function testIsUniqueMultipleFieldsOneIsNull()
     }
 
     /**
+     * Tests the IsNotReferencedBy rule before delete
+     *
+     * @group delete
+     * @return void
+     */
+    public function testIsNotReferencedByRule()
+    {
+        $articlesTable = TableRegistry::get('Articles');
+        $articlesTable->belongsTo('Authors');
+        $authorsTable = TableRegistry::get('Authors');
+
+        $author = $authorsTable->get(1);
+        $authorWithoutArticles = $authorsTable->get(2);
+
+        $rules = $authorsTable->rulesChecker();
+        $rules->addDelete($rules->IsNotReferencedBy('Articles'));
```

Comment:
@markstory of course I understand. Just seemed to me, like there was a lot of redundant code, which would grow with each Rule added. (Doesn't seem to me though, like anyone is planning on adding extra rules anytime soon :) )